data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/DataRequest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.model;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+/**
+ * The FilteredResourceRequest is the input for results-service where the resource is queued-up ready for the client's request
+ * to retrieve the Resource.
+ * TopicOffsetMessage is the output for this service which will be send the client the information needed to
+ * retrieve this data.
+ * Note there are two classes that effectively represent the same data but represent a different stage of the process.
+ * uk.gov.gchq.palisade.service.attributemask.model.AttributeMaskingResponse is the output from the attribute-masking-service.
+ * uk.gov.gchq.palisade.service.filteredresource.model.FilteredResourceRequest is the input for the filtered-resource-service.
+ */
```

Comment:
I'm not sure that this JavaDoc is relevant to the DataRequest. Why is it referring to the **input** for the `filtered-resource-service` and the **output** of the `topic-offset-service`?

Reply:
Resolved